ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -0,0 +1,1072 @@
+        double xa = lowPositionRho * Math.cos(lowPositionAlpha);
+        double ya = lowPositionRho * Math.sin(lowPositionAlpha);
+        double xb = highPositionRho * Math.cos(highPositionAlpha);
+        double yb = highPositionRho * Math.sin(highPositionAlpha);
+        return Math.toDegrees(Math.atan((yb - ya) / (xb - xa)));
+    }
+
+    /**
+     * @param twoWindingsTransformer The twoWindingsTransformer containing the PhaseTapChanger we want to convert
+     * @return P (MW) of the angle regulation for the two windings transformer
+     */
+    private float calculateAngleP(TwoWindingsTransformer twoWindingsTransformer) {
+        return (float) twoWindingsTransformer.getPhaseTapChanger().getRegulationValue();
+    }
+
+    /**
+     * Give the type of the UcteAngleRegulation
+     *
+     * @param twoWindingsTransformer containing the PhaseTapChanger we want to convert
+     * @return The type of the UcteAngleRegulation
+     */
+    private UcteAngleRegulationType findRegulationType(TwoWindingsTransformer twoWindingsTransformer) {
+        if (isSymm(twoWindingsTransformer)) {
+            return UcteAngleRegulationType.SYMM;
+        } else {
+            return UcteAngleRegulationType.ASYM;
+        }
+    }
+
+    private boolean isSymm(TwoWindingsTransformer twoWindingsTransformer) {
+        for (int i = twoWindingsTransformer.getPhaseTapChanger().getLowTapPosition();
+             i < twoWindingsTransformer.getPhaseTapChanger().getHighTapPosition(); i++) {
+            if (twoWindingsTransformer.getPhaseTapChanger().getStep(i).getRho() != 1) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    UcteNodeCode createUcteNodeCode(String id, VoltageLevel voltageLevel, String country) {
+        UcteNodeCode ucteNodeCode;
+        if (iidmIdToUcteNodeCodeId.containsKey(id)) {
+            return iidmIdToUcteNodeCodeId.get(id);
+        }
+        Optional<UcteNodeCode> optionalUcteNodeCode = parseUcteNodeCode(id);
+        if (optionalUcteNodeCode.isPresent()) { // the ID is already an UCTE id
+            iidmIdToUcteNodeCodeId.put(id, optionalUcteNodeCode.get());
+            return optionalUcteNodeCode.get();
+        } else {
+            convertIidmIdToUcteNodeCode(id, voltageLevel, country);
+            ucteNodeCode = iidmIdToUcteNodeCodeId.get(id);
+        }
+        return ucteNodeCode;
+    }
+
+    /**
+     * create a unique UcteNodeCode using the voltage level and the country given and
+     * store it to {@link UcteExporter#iidmIdToUcteNodeCodeId} so we don't need to create it every time
+     * and ensure  its uniqueness
+     *
+     * @param id the orignal id of the iidm component we want to convert
+     * @param voltageLevel the {@link VoltageLevel} of the component we want to convert
+     * @param country the string corresponding to the country where the iidm component is
+     */
+    void convertIidmIdToUcteNodeCode(String id, VoltageLevel voltageLevel, String country) {
```

Comment:
Should this method be private.
Why don't you pass a real Country object instead of a String?

Reply:
To be discussed: do we remove this method?